### PR TITLE
[Storage] STG87

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR merges the STG87 branches into `main`. This release contains the following features:

- DataLake Encryption Context
- File-Share Trailing Dot Support
- File-Share Oauth Support

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/CHANGELOG.md [hunk 1]
```diff
@@ -3,16 +3,22 @@
 ## 12.12.0b1 (Unreleased)
 
 ### Features Added
-
-### Bugs Fixed
+- Added support for service version 2022-11-02.
+- Added support for `TokenCredential` to be used for authentication. A `TokenCredential` can be provided for the
+`credential` parameter to any client constructor. **Note:** When using a `TokenCredential`, the new keyword parameter
+`token_intnet` is **required** and must be provided. Addtitionally, this form of authentication is only supported for
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-file-share/CHANGELOG.md:9]: 👀

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**